tests/test_load_corrector.py

```diff
@@ -0,0 +1,187 @@
+
+    # Create time range and new data for 14 days (2 weeks)
+    time_range = pd.date_range(start="2023-09-25", periods=24 * 14, freq="H")
+
+    # Create new measured_data and predicted_data matching the time range
+    measured_data = pd.DataFrame(
+        {
+            "time": time_range,
+            "Last": np.random.rand(len(time_range)) * 100,  # Random 'Last' values
+        }
+    )
+
+    predicted_data = pd.DataFrame(
+        {
+            "time": time_range,
+            "Last Pred": np.random.rand(len(time_range)) * 100,  # Random 'Last Pred' values
+        }
+    )
+
+    adjuster = LoadPredictionAdjuster(measured_data, predicted_data, load_forecast)
+    adjuster.merged_data = adjuster._merge_data()
+
+    # Calculate the weighted mean over training and testing periods
+    adjuster.calculate_weighted_mean(train_period_weeks=1, test_period_weeks=1)
+
+    # Assert that weekday and weekend differences are calculated and non-empty
+    assert adjuster.weekday_diff is not None, "Weekday differences should be calculated"
+    assert adjuster.weekend_diff is not None, "Weekend differences should be calculated"
+    assert len(adjuster.weekday_diff) > 0, "Weekday differences should not be empty"
+    assert len(adjuster.weekend_diff) > 0, "Weekend differences should not be empty"
+
+
+def test_adjust_predictions(
+    setup_data: tuple[pd.DataFrame, pd.DataFrame, MagicMock],
+) -> None:
+    """
+    Test the adjust_predictions method to ensure it correctly adds the 'Adjusted Pred' column to train and test data.
+    """
+    measured_data, predicted_data, load_forecast = setup_data
+    adjuster = LoadPredictionAdjuster(measured_data, predicted_data, load_forecast)
+    adjuster.merged_data = adjuster._merge_data()
+
+    # Calculate the weighted mean and adjust predictions
+    adjuster.calculate_weighted_mean(train_period_weeks=1, test_period_weeks=1)
+    adjuster.adjust_predictions()
+
+    # Assert that the 'Adjusted Pred' column is present in both train and test data
+    assert (
+        "Adjusted Pred" in adjuster.train_data.columns
+    ), "Train data should have 'Adjusted Pred' column"
+    assert (
+        "Adjusted Pred" in adjuster.test_data.columns
+    ), "Test data should have 'Adjusted Pred' column"
+
+
+def test_evaluate_model(
+    setup_data: tuple[pd.DataFrame, pd.DataFrame, MagicMock],
+    capsys: pytest.CaptureFixture,
+) -> None:
+    """
+    Test the evaluate_model method to ensure it prints evaluation metrics (MSE and R-squared).
+    """
+    measured_data, predicted_data, load_forecast = setup_data
+    adjuster = LoadPredictionAdjuster(measured_data, predicted_data, load_forecast)
+    adjuster.merged_data = adjuster._merge_data()
+
+    # Calculate weighted mean, adjust predictions, and evaluate the model
+    adjuster.calculate_weighted_mean(train_period_weeks=1, test_period_weeks=1)
+    adjuster.adjust_predictions()
+    adjuster.evaluate_model()
```

Comment:
With the suggested returns above this can be simplified

```suggestion
    mse, r2 = adjuster.evaluate_model()
    assert not np.isnan(mse)
    assert not np.isnan(r2)
```